Source/WTF/wtf/Dominators.h

```diff
@@ -294,6 +314,129 @@ class Dominators {
+                    m_order.append(index);
+                    continue;
+                }
+                workList.append(~index); // Revisit this index later to add it after visiting all successors.
+                for (unsigned i = 0; i < successors.size(); i ++) {
+                    int16_t index = m_graph.index(successors[i]);
+                    if (!visited.get(index)) {
+                        visited.set(index);
+                        workList.append(index);
+                    }
+                }
+            }
+            m_postorderNumbers.fill(0, m_graph.numNodes());
+            for (unsigned i = 0; i < m_order.size(); i ++)
+                m_postorderNumbers[m_order[i]] = i;
+            std::reverse(m_order.begin(), m_order.end());
+        }
+
+        uint16_t intersect(uint16_t a, uint16_t b)
+        {
+            while (a != b) {
+                while (m_postorderNumbers[a] < m_postorderNumbers[b])
+                    a = m_idoms[a];
+                while (m_postorderNumbers[b] < m_postorderNumbers[a])
+                    b = m_idoms[b];
+            }
+            return a;
+        }
+
+        void compute()
+        {
+            computeReversePostorder();
+
+            bool fixpoint = false;
+            int16_t rootIndex = m_graph.index(m_graph.root());
+            m_idoms[rootIndex] = rootIndex;
+            ASSERT(m_order[0] == rootIndex);
+            while (!fixpoint) {
+                fixpoint = true;
+                for (unsigned i = 1; i < m_order.size(); i ++) {
+                    uint16_t node = m_order[i];
+                    uint16_t newIdom = m_idoms[node];
+                    bool isFirstProcessed = true;
+                    const typename Graph::Node& block = m_graph.node(node);
+                    for (auto pred : m_graph.predecessors(block)) {
+                        uint16_t predIndex = m_graph.index(pred);
+                        uint16_t predIdom = m_idoms[predIndex];
+                        if (predIdom == undefinedIdom)
+                            continue;
+                        if (isFirstProcessed) {
+                            newIdom = predIndex;
+                            isFirstProcessed = false;
+                        } else
+                            newIdom = intersect(newIdom, predIndex);
+                    }
+                    if (m_idoms[node] != newIdom) {
+                        ASSERT(newIdom != undefinedIdom);
+                        fixpoint = false;
+                        m_idoms[node] = newIdom;
+                    }
+                }
+            }
+        }
+
+        typename Graph::Node immediateDominator(typename Graph::Node block)
+        {
+            if (block == m_graph.root())
+                return nullptr;
+            return m_graph.node(m_idoms[m_graph.index(block)]);
+        }
+
+    private:
+        Graph& m_graph;
+        Vector<uint16_t, 64> m_idoms;
+        Vector<uint16_t, 64> m_order;
```

Comment:
Maybe, naming `m_reversePostorderedNodes`?

Reply:
Seems reasonable to me.